R/zzz.R

```diff
@@ -1,3 +1,5 @@
 .onLoad <- function(libname, pkgname) { # nolint
   teal.logger::register_logger(namespace = "teal.modules.general")
 }
+
+ggplot_themes <- c("gray", "bw", "linedraw", "light", "dark", "minimal", "classic", "void")
```

Comment:
Hey, this PR is excellent and removes a lot of repetitions in the code. I like the idea of having a variable storing all themes that we reuse. The only, tiny, hesitance about this one is the possibility that a variable called `ggplot_themes` already exists in someone's session when tmg is loaded. Then there is a conflict.

What if (pick one, my preference is the `options`)
- we change the name from `ggplot_themes` to `teal_ggplot_themes`
- or we set an option `options(teal_ggplot_themes = c("gray", "bw", "linedraw", "light", "dark", "minimal", "classic", "void"))` and we use `getOpion('teal_ggplot_themes')` in `selectInput`
- we create a hidden environment called `.teal_transform_options <- env()` where we assign `ggplot_themes` and then we use `get('ggplot_themes', env = '.teal_transform_options')` in `selectInput`

Reply:
`teal_ggplot_themes` would indicate that these themes are somehow connected to or defined in `teal`, which is not the case: it is merely an enumeration of themes provided by `ggplot2`.

Functions will scope their enclosing environment first before scoping the global environment. See here:
```
library(teal)
devtools::load_all()

ggplot_themes <- c("gray", "void")

data <- teal_data()
data <- within(data, {
  library(nestcolor)
  ADSL <- teal.modules.general::rADSL
})
datanames <- c("ADSL")
datanames(data) <- datanames
join_keys(data) <- default_cdisc_join_keys[datanames]
app <- teal::init(
  data = data,
  modules = teal::modules(
    teal.modules.general::tm_a_pca(
      "PCA",
      dat = teal.transform::data_extract_spec(
        dataname = "ADSL",
        select = teal.transform::select_spec(
          choices = teal.transform::variable_choices(
            data = data[["ADSL"]], c("BMRKR1", "AGE", "EOSDY")
          ),
          selected = c("BMRKR1", "AGE"),
          multiple = TRUE
        ),
        filter = NULL
      ),
      ggplot2_args = teal.widgets::ggplot2_args(
        labs = list(subtitle = "Plot generated by PCA Module")
      )
    )
  )
)
runApp(app, launch.browser = TRUE)
```

Reply:
My bad, I though `ggplot_themes` will be assigned to `.GlobalEnv()` when this package is loaded. It's gonna be just in tmg namespace, so it will not pollute global environment. Sorry for the confusion. Alrighty so I think this is good to go